Marlin/ultralcd.cpp

```diff
@@ -711,15 +711,44 @@ void kill_screen(const char* lcd_msg) {
 
   #if ENABLED(MENU_ITEM_CASE_LIGHT)
 
+    extern uint8_t case_light_brightness;
     extern bool case_light_on;
     extern void update_case_light();
+    extern bool useable_hardware_PWM(uint8_t);
+    int case_light_brightness_int;
 
-    void toggle_case_light() {
-      case_light_on ^= true;
-      lcdDrawUpdate = LCDVIEW_KEEP_REDRAWING;
+    void level_case_light() {
+      case_light_brightness = (uint8_t) case_light_brightness_int;
+      update_case_light();
+    }
+
+    void on_case_light() {
+      case_light_on = true;
+      update_case_light();
+    }
+
+    void off_case_light() {
+      case_light_on = false;
       update_case_light();
     }
 
+    void case_light_menu() {
+      START_MENU();
+      //
+      // ^ Main
+      //
+      MENU_BACK(MSG_MAIN);
+
+      if (useable_hardware_PWM(CASE_LIGHT_PIN)) {
+        case_light_brightness_int = case_light_brightness;
+        MENU_ITEM_EDIT_CALLBACK(int3, MSG_CASE_LIGHT_BRIGHTNESS, &case_light_brightness_int, 0, 255, level_case_light, true);
+      }
+      MENU_ITEM(function, MSG_CASE_LIGHT_ON, on_case_light);
+      MENU_ITEM(function, MSG_CASE_LIGHT_OFF, off_case_light);
```

Comment:
Actually, this is even better:
```cpp
    MENU_ITEM_EDIT_CALLBACK(bool, MSG_CASE_LIGHT, &case_light_on, update_case_light);
```
…then no other functions are needed for toggling.